support-workers/src/main/scala/com/gu/zuora/subscriptionBuilders/SubscriptionBuilder.scala

```diff
@@ -0,0 +1,94 @@
+package com.gu.zuora.subscriptionBuilders
+
+import cats.data.EitherT
+import cats.implicits._
+import com.gu.support.config.{TouchPointEnvironment, ZuoraContributionConfig}
+import com.gu.support.promotions.{PromoError, PromotionService}
+import com.gu.support.redemption.{InvalidCode, InvalidReaderType}
+import com.gu.support.redemptions.RedemptionData
+import com.gu.support.workers._
+import com.gu.support.workers.states.CreateZuoraSubscriptionState
+import com.gu.support.zuora.api.ReaderType.Gift
+import com.gu.support.zuora.api.{ReaderType, SubscriptionData}
+import com.gu.zuora.subscriptionBuilders.ProductSubscriptionBuilders.buildContributionSubscription
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+case class BuildSubscribePromoError(cause: PromoError) extends RuntimeException
+
+case class BuildSubscribeRedemptionError(cause: InvalidCode) extends RuntimeException
+
+class SubscriptionBuilder(
+  digitalSubscriptionPurchaseBuilder: DigitalSubscriptionPurchaseBuilder,
+  digitalSubscriptionCorporateRedemptionBuilder: DigitalSubscriptionCorporateRedemptionBuilder,
+  promotionService: => PromotionService,
+  config: BillingPeriod => ZuoraContributionConfig,
+) {
+
+  def build(
```

Comment:
this replaces both the original >50lines function that was in CreateZuoraSubscription.scala originally